File: src/com/sterndu/json/ShortValue.java
package com.sterndu.json;

import java.util.function.Function;

public class ShortValue implements NumberValue {

	private short value = 0;

	public ShortValue(short value) {
		this.value = value;
	}

	@Override
	public Number getValue() { return value; }

	public short getValueShort() { return value; }

	public void setValue(short value) { this.value = value; }

	@Override
	public String toJson() {
		return value + "";
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return value + "";
	}

	@Override
	public JsonValue toJsonValue() {
		return this;
	}

	@Override
	public JsonValue toJsonValue(Function<Object, String> function) {
		return this;
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/ShortValue.kt
@file:JvmName("ShortValue")
package com.sterndu.json

class ShortValue @JvmOverloads constructor(value: Short = 0) : NumberValue {

	private var valueShort: Short = value

	fun getValueShort(): Short {
		return valueShort
	}

	override fun getValue(): Number {
		return valueShort
	}

	fun setValue(value: Short) {
		valueShort = value
	}

	override fun toJson(): String {
		return valueShort.toString() + ""
	}

	override fun toJson(function: Function1<Any?, String>): String {
		return valueShort.toString() + ""
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: Function1<Any?, String>): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}
}
